### PR TITLE
[JAMES-2999] reuse spamassassin images between runs

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinReuseCache.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.spamassassin;
+
+public class SpamAssassinReuseCache {
```
Comment (rouazana) [inline third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinReuseCache.java]: what's the point of this empty unused class?

Comment (mbaechler) [reply third-party/spamassassin/src/test/java/org/apache/james/spamassassin/SpamAssassinReuseCache.java]: this is a mistake

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
> Can you provide build time gain rational please?

The build log is gone so having CI figures will require yet another run.
However, this changeset is not related to CI, it's a enhancement for the developer as it allows to run tests that require spamassassin much faster.
I don't expect any CI time loss as in worst case it will just do as before.
The only risk is having many images filling the CI disk space but as every image are using the exact same layer, they are just some metadata pointing to already existing data.

Do you want a new CI run to check the build time?

**Comment by chibenwa** [comment]
No that's ok ^^

I was just curious.

**Comment by chibenwa** [comment]
Was :green_apple:

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**